### PR TITLE
I'm the master of disguise

### PR DESCRIPTION
**Pull request by kashike**
Adds #130.
Supersedes #158.

### CODE REVIEW

File: src/test/java/org/kitteh/irc/client/library/util/mask/MaskTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        final HostMask mask = HostMask.fromHost("kitten.institute");
+
+        Assert.assertEquals("kitten.institute", mask.getHost());
+        Assert.assertEquals("*!*@kitten.institute", mask.asString());
+
+        assertContainsOnly(mask, Arrays.asList(
+            new TestUser("mbaxter", "~mbax", "kitten.institute", null),
+            new TestUser("kashike", "kashike", "kitten.institute", null)
+        ), Collections.singletonList(
+            new TestUser("lol768", "lol768", "andy.in.toy.story", null)
```
Comment (lol768) [inline src/test/java/org/kitteh/irc/client/library/util/mask/MaskTest.java]: 😐

**Comment by bendem** [comment]
I'm not quite sure why you added `Mask.AsString extends Mask` since you're only using `AsString`. I'd move that method to `Mask` and use that instead. I can't think of a mask that's not representable as a string anyway

**Comment by bendem** [comment]
Partial wildcards are not supported either. A HostMask with host = `192.168.*` will not work.

**Comment by mbax** [comment]
https://github.com/KittehOrg/KittehIRCClientLib/issues/130#issuecomment-465359071

**Pull request closed without merging**